File: backend/app/models/bank_account.py
import uuid
from sqlalchemy import Column, String, DateTime, ForeignKey, func
from app.database import Base


class BankAccount(Base):
    __tablename__ = "bank_accounts"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    merchant_id = Column(String, ForeignKey("merchants.id"), nullable=False, index=True)
    bank_name = Column(String, nullable=True)
    routing_number = Column(String, nullable=False)
    encrypted_account_number = Column(String, nullable=False)
    account_type = Column(String, default="checking")  # checking, savings
    verification_status = Column(String, default="pending")  # pending, micro_deposit_sent, verified, failed
    micro_deposit_amount_1 = Column(String, nullable=True)
    micro_deposit_amount_2 = Column(String, nullable=True)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

File: mobile_flutter/lib/models/bank_account.dart
class BankAccount {
  final String id;
  final String merchantId;
  final String? bankName;
  final String routingNumber;
  final String? accountNumberLast4;
  final String accountType;
  final String verificationStatus;
  final String? microDepositAmount1;
  final String? microDepositAmount2;
  final DateTime? createdAt;

  BankAccount({
    required this.id,
    required this.merchantId,
    this.bankName,
    required this.routingNumber,
    this.accountNumberLast4,
    required this.accountType,
    required this.verificationStatus,
    this.microDepositAmount1,
    this.microDepositAmount2,
    this.createdAt,
  });

  factory BankAccount.fromJson(Map<String, dynamic> json) {
    return BankAccount(
      id: json['id'] as String,
      merchantId: json['merchant_id'] as String,
      bankName: json['bank_name'] as String?,
      routingNumber: json['routing_number'] as String,
      accountNumberLast4: json['account_number_last4'] as String?,
      accountType: json['account_type'] as String,
      verificationStatus: json['verification_status'] as String,
      microDepositAmount1: json['micro_deposit_amount_1'] as String?,
      microDepositAmount2: json['micro_deposit_amount_2'] as String?,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'merchant_id': merchantId,
        'bank_name': bankName,
        'routing_number': routingNumber,
        'account_number_last4': accountNumberLast4,
        'account_type': accountType,
        'verification_status': verificationStatus,
      };
}
